Fix: ValueError handling for empty files scanning with only-allowlist…

Roman-Malinowski opened the pull request:
Fixes #936 

- [X] All CI checks are green

* **What kind of change does this PR introduce?**
<!-- (Bug fix, feature, docs update, ...) -->
Bug fix: Handling the ValueError when running `detect-secrets scan --only-allowlisted` in a directory containing empty files

* **What is the current behavior?**
<!-- (You can also link to an open issue here) -->
A ValueError is raised as described in [issue #936 ](https://github.com/Yelp/detect-secrets/issues/936)

* **What is the new behavior (if this is a feature change)?**
A warning is raised, and the scan continues.

Roman-Malinowski committed:
Fix: ValueError handling for empty files scanning with only-allowlisted flag

tocic commented on detect_secrets/core/scan.py:
```diff
@@ -206,6 +206,9 @@ def scan_for_allowlisted_secrets_in_file(filename: str) -> Generator[PotentialSe
         return
+    except ValueError:
+        log.warning(f'Unable to scan file: {filename}. Please ignore if file is empty.')
+        return
```
I would suggest checking `if lines:` above instead - that way we'll be specifically addressing the issue (as opposed to any `ValueError` exception) and we won't need to execute the line `yield from _scan_for_allowlisted_secrets_in_lines(enumerate(lines, start=1), filename)` at all

Roman-Malinowski committed:
Replaced try/except with length test

Update scanning logic to check if lines are not empty before scanning for allowlisted secrets.
